src/lib/support/tests/TestZclString.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements a unit test suite for ZclString
+ *
+ */
+
+#include <inttypes.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <nlunit-test.h>
+#include <support/CHIPMem.h>
+#include <support/CodeUtils.h>
+#include <support/UnitTestRegistration.h>
+
+#include <lib/support/ZclString.h>
+
+using namespace chip;
+using namespace chip::Logging;
+using namespace chip::Platform;
+
+static void TestZclString64Chars(nlTestSuite * inSuite, void * inContext)
+{
+    // Init zcl string
+    uint8_t zclString[255];
+    // Init c string
+    char * cString64 = static_cast<char *>(MemoryCalloc(1024, true));
+    // Add random chars to c string
+    for (size_t i = 0; i < 64; i++)
+        cString64[i] = 'A';
+
+    MakeZclCharString(zclString, cString64);
+    int charIsTerminated = 0;
+    for (int i = 0; i != 256; i++)
+        if (zclString[i] == '\0')
+        {
+            charIsTerminated = 1;
+            break;
+        }
+
+    // Memmory assert
+    NL_TEST_ASSERT(inSuite, cString64 != nullptr);
+    // ZCL String has char terminated
+    NL_TEST_ASSERT(inSuite, charIsTerminated == 1);
+    // ZCL String length assert
+    NL_TEST_ASSERT(inSuite, zclString[0] == 64);
+
+    chip::Platform::MemoryFree(cString64);
+}
+
+static void TestZclString254Chars(nlTestSuite * inSuite, void * inContext)
+{
+    // Init zcl string
+    uint8_t zclString[255];
+    // Init c string
+    char * cString254 = static_cast<char *>(MemoryCalloc(1024, true));
+    // Add random chars to c string
+    for (size_t i = 0; i < 254; i++)
+        cString254[i] = 'A';
+
+    MakeZclCharString(zclString, cString254);
+    int charIsTerminated = 0;
+    for (int i = 0; i != 256; i++)
+        if (zclString[i] == '\0')
+        {
+            charIsTerminated = 1;
+            break;
+        }
+
+    // Memmory assert
+    NL_TEST_ASSERT(inSuite, cString254 != nullptr);
+    // ZCL String has char terminated
+    NL_TEST_ASSERT(inSuite, charIsTerminated == 1);
```

Comment:
Again, I don't see why this would necessarily be 1.

Reply:
Removed these tests.